server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -187,6 +187,11 @@ public KeywordFieldMapper build(BuilderContext context) {
         }
     }
 
+    @Override
+    protected String parseSourceValue(Object value) {
+        return value.toString();
```

Comment:
Keywords shouldn't also consider `ignore_above` and the extraction from `_source` to happen with this value in mind?

Reply:
I'm planning to tackle `ignore_above` (and likely `null_value` too) in a follow-up PR.